src/main/java/org/jabref/gui/referencemetadata/ReferenceMetadataFetcherGoogleScholar.java

```diff
@@ -0,0 +1,431 @@
+                                Localization.lang("Please show Google Scholar, that you are not a robot, by opening the link") + "\n\n" +
+                                        "https://scholar.google.com/scholar?q=google \n\n" +
+                                        Localization.lang("and solving the shown captcha, otherwise the reference metadata cannot be fetched."),
+                                Localization.lang("Continue"),
+                                Localization.lang("Cancel"));
+
+                        if (result == 1) {
+                            retryFetchingMetadata = true; // retry fetching metadata
+                            break;
+                        } else {
+                            entriesWithIncompleteMetadata.addAll(ReferenceMetadataUtils.getAllEntriesStartingWithGivenIndex(startIndexEntriesBlock + rxEntryIndex, entries));
+                            return true; // cancel fetching metadata
+                        }
+                    } else if (_status_tooManyRequests) {
+                        LOGGER.info("too many requests: Google Scholar asks you to wait some time before " +
+                                "sending further requests");
+
+                        int result = showCustomDialogAndWait(dialogService, Alert.AlertType.INFORMATION,
+                                Localization.lang("Too Many Requests"),
+                                Localization.lang("Google Scholar asks you to wait some time before sending further requests."),
+                                Localization.lang("Continue"),
+                                Localization.lang("Cancel"));
+
+                        if (result == 1) {
+                            retryFetchingMetadata = true; // retry fetching metadata
+                            break;
+                        } else {
+                            entriesWithIncompleteMetadata.addAll(ReferenceMetadataUtils.getAllEntriesStartingWithGivenIndex(startIndexEntriesBlock + rxEntryIndex, entries));
+                            return true; // cancel fetching metadata
+                        }
+                    }
+
+                    // find entry to update
+                    BibEntry entry = getBibEntryWithGivenEntryId(database, _entryId);
+
+                    if (entry == null) {
+                        LOGGER.info("skipping bibEntry, since entry with entryId=" + _entryId + " could not " +
+                                "be found");
+
+                        continue;
+                    }
+
+                    // set (updated) entry data (citation count)
+                    entry.setField(InternalField.CITATION_COUNT, citationCount);
+
+                    if (UPDATE_NOTE_FIELD_WITH_CITATION_COUNT) {
+                        entry.setField(StandardField.NOTE, note);
+                    }
+                }
+            }
+
+            if (ENABLE_RANDOM_DWELL_TIME) {
```

Comment:
It's some potential measure. It may be useful. Since the queries are performed in a real browser, less queries get blocked and adding random dwell times would be as if a real user presses `F5` subsequently (resulting in different time intervals).